### PR TITLE
feat: implement config on `controller`

### PR DESCRIPTION
**Pull request by Roxxas96**
## Overall explanation of your work :

I implemented a config manager to prevent having constant variables hardcoded in the `main.rs`.

## Technical decisions you've made and why :

- `confy` is a simple crate to load config from file or env. variables.

## Tasks lists :

- [x] Implement `controller` config

### CODE REVIEW

File: controller/src/main.rs [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
-    let grpc_server_addr: String = "0.0.0.0:50051".to_string();
-    let grpc_server_num_workers: usize = 1;
-
-    let http_server_addr: String = "0.0.0.0:3000".to_string();
-    let http_server_num_workers: usize = 1;
-
     // Init Logger
     env_logger::init();
 
+    let config: config::KudoControllerConfig = confy::load_path("controller.conf").unwrap();
```
Comment (sameo) [inline controller/src/main.rs:11]: You should return a result here.

**Review by sameo — COMMENTED** [review]
LGTM, except for a nitty `unwrap`. Please fix it and we'll merge that PR.

File: controller/src/main.rs (hunk 1 above)
Comment (Roxxas96) [reply controller/src/main.rs]: But this one result in the same consequence as it is in the main function, right ? (I changed it anyway directly by doing a force push on commit d8bf76e)

Comment (sameo) [reply controller/src/main.rs]: Consequences are different. `unwrap` panics, while a main returning an error is handled more gracefully (proper stack dump, etc).

File: controller/lib/src/internal_api/interface.rs [hunk 2]
```diff
@@ -15,7 +15,6 @@ impl InternalAPIInterface {
         );
 
         for _ in 1..num_workers {
-            let address = address.clone();
```
Comment (sameo) [inline controller/lib/src/internal_api/interface.rs:16]: Merge that change into the previous commit, please.

Comment (Roxxas96) [reply controller/lib/src/internal_api/interface.rs:16]: Done !

**Review by sameo — APPROVED** [review]

**Pull request merged**